extensions/amp-a4a/amp-a4a-format.md

```diff
@@ -0,0 +1,429 @@
+
+**Good**
+```css
+.amp-animate .box {
+    transform: rotate(180deg);
+    transition: transform 2s;
+}
+```
+
+**Bad**
+
+Property not allowed in CSS class.
+```css
+.amp-animate .box {
+    color: red;  // non-animation property not allowed in animation selector
+    transform: rotate(180deg);
+    transition: transform 2s;
+}
+```
+
+Missing context class `.amp-animate`.
+```css
+.box {
+    transform: rotate(180deg);
+    transition: transform 2s;
+}
+```
+
+##### Transitionable and animatable properties
+
+The only properties that may be transitioned are opacity and transform.
+([Rationale](http://www.html5rocks.com/en/tutorials/speed/high-performance-animations/))
+
+**Good**
+```css
+transition: transform 2s;
+```
+
+**Bad**
+```css
+transition: background-color 2s;
+```
+
+**Good**
+```css
+@keyframes turn {
+  from {
+    transform: rotate(180deg);
+  }
+  
+  to {
+    transform: rotate(90deg);
+  }
+}
+```
+
+**Bad**
+```css
+@keyframes slidein {
+  from {
+    margin-left:100%;
+    width:300%
+  }
+  
+  to {
+    margin-left:0%;
+    width:100%;
+  }
+}
+```
+
+
+### AMP Extensions
+
+The following are _allowed_ AMP extension modules in an A4A creative. Extensions
+not explicitly allowed are prohibited.
+
+Most of the omissions are either for performance or to make A4A creatives 
+simpler to analyze.
+
+_Example:_ `<amp-ad>` is omitted from this list.  It is explicitly disallowed
+because allowing an `<amp-ad>` inside an `<amp-ad>` could potentially lead to
+unbounded waterfalls of ad loading, which does not meet A4A performance goals.
+
+_Example:_ `<amp-iframe>` is omitted from this list.  It is disallowed 
+because ads could use it to execute arbitrary Javascript and load arbitrary 
+content. Ads wanting to use such capabilities should return `false` from 
+their
+[a4aRegistry](https://github.com/ampproject/amphtml/blob/master/ads/_a4a-config.js#L40)
+entry and use the existing '3p iframe' ad rendering mechanism.
+
+_Example:_ `<amp-ad-network-*-impl>` are omitted from this list.  The 
+`<amp-ad>` tag handles delegation to these implementation tags; creatives 
+should not attempt to include them directly.
+
+<table>
+  <tr><td>amp-accordion</td></tr>
+  <tr><td>amp-analytics</td></tr>
+  <tr><td>amp-anim</td></tr>
+  <tr><td>amp-audio</td></tr>
+  <tr><td>amp-brid-player</td></tr>
+  <tr><td>amp-brightcove</td></tr>
+  <tr><td>amp-carousel</td></tr>
+  <tr><td>amp-dailymotion</td></tr>
+  <tr><td>amp-facebook</td></tr>
+  <tr><td>amp-fit-text</td></tr>
+  <tr><td>amp-font</td></tr>
+  <tr><td>amp-form</td></tr>
+  <tr><td>amp-fx-flying-carpet</td></tr>
+  <tr><td>amp-google-vrview-image</td></tr>
+  <tr><td>amp-image-lightbox</td></tr>
+  <tr><td>amp-instagram</td></tr>
+  <tr><td>amp-jwplayer</td></tr>
+  <tr><td>amp-kaltura-player</td></tr>
+  <tr><td>amp-lightbox</td></tr>
+  <tr><td>amp-list</td></tr>
+  <tr><td>amp-live-list</td></tr>
+  <tr><td>amp-o2-player</td></tr>
+  <tr><td>amp-pinterest</td></tr>
```

Comment:
Oh wait -- `amp-pixel` and `amp-video` are not technically extensions -- they're builtins.  That said, it seems worth including them in this list b/c they're not covered anywhere else.  @Gregable let me know if you have objections to that.

`amp-experiment` was missed b/c I don't think it existed when this list was first formulated.

All three added.